tests/test_operations.py

```diff
@@ -303,127 +304,154 @@ def test_secrets_crud(self):
-                    secrets.get_secret(
-                        [], argparse.Namespace(secret_names=[testvar_name], json=False)
-                    )
-                output = "".join(output)
-                self.assertIn(testvar_name, output)
-                self.assertIn(testvar_value, output)
-                # Multiple variables:
-                with CaptureStdout() as output:
-                    secrets.get_secret(
-                        [],
-                        argparse.Namespace(
-                            secret_names=[testvar_name, unusedvar_name],
-                            json=False
-                        )
-                    )
-                output = "".join(output)
-                self.assertIn(testvar_name, output)
-                self.assertIn(testvar_value, output)
-                #
-                # Now JSON get call:
-                # Single variable:
-                with CaptureStdout() as output:
+
+                # New output file
+                tempfile = '.dss-test-operations-temp-output'
+                if os.path.exists(tempfile):
+                    subprocess.call(['rm', '-f', tempfile])
+                secrets.get_secret(
+                    [], argparse.Namespace(secret_name=testvar_name, outfile=tempfile, force=False)
+                )
+                with open(tempfile, 'r') as f:
+                    file_contents = f.read()
+                self.assertIn(testvar_value, file_contents)
+
+                # Output file already exists, use force
+                with self.assertRaises(RuntimeError):
                     secrets.get_secret(
-                        [],
-                        argparse.Namespace(secret_names=[testvar_name], json=True),
+                        [], argparse.Namespace(secret_name=testvar_name, outfile=tempfile, force=False)
                     )
-                output = "".join(output)
-                self.assertIn(testvar_name, output)
-                self.assertIn(testvar_value, output)
-                # Multiple variables:
+                secrets.get_secret(
+                    [], argparse.Namespace(secret_name=testvar_name, outfile=tempfile, force=True)
+                )
+
+                # Stdout
                 with CaptureStdout() as output:
                     secrets.get_secret(
-                        [],
-                        argparse.Namespace(
-                            secret_names=[testvar_name, unusedvar_name],
-                            json=True
-                        )
+                        [], argparse.Namespace(secret_name=testvar_name, outfile=None, force=False)
                     )
-                output = "".join(output)
-                self.assertIn(testvar_name, output)
-                self.assertIn(testvar_value, output)
+                self.assertIn(testvar_value, "\n".join(output))
+
+                # Clean up
+                subprocess.call(['rm', '-f', tempfile])
```

Comment:
Nit: consider `os.remove()`/`os.unlink()` instead of `subprocess.call(['rm', '-f', tempfile])`. It shouldn't have the extra overhead of `subprocess`, and is cross-platform compatible.

Also, `check_call` will wait until the file is deleted.  `call` will just initiate the command and move to the next bit of code regardless of whether the process succeeded or even started.

Reply:
using tempfile instead